packages/concerto-core/lib/decoratormanager.js

```diff
@@ -474,6 +654,38 @@ class DecoratorManager {
             this.applyDecorator(property, type, decorator);
         }
     }
+    /**
+     * extracts a Command from a ClassDeclaration or its properties, as required.
+     * @param {String} namespace the namespace for the declaration
+     * @param {*} declaration the class declaration
+     * @param {Object} decoratorDict the dictionary with extracted decorators
+     * @param {boolean} shouldRemoveDecorators whether to remove decorators from base model
+     */
+    static extractDecoratorFromModel(namespace, declaration, decoratorDict, shouldRemoveDecorators) {
+        this.processDecorators(namespace, declaration.decorators, decoratorDict, shouldRemoveDecorators,declaration.name,'');
+        if (declaration.properties) {
+            declaration.properties.forEach((property) => {
+                this.processDecorators(namespace, property.decorators, decoratorDict, shouldRemoveDecorators,declaration.name,property.name);
+            });
+        }
+    }
+    /**
+    * Process decorators for a given set of decorators.
+    * @param {string} namespace - The namespace to associate with the decorators.
+    * @param {Array<string>} decorators - The array of decorators to process.
+    * @param {Object} decoratorDict - The decorator dictionary to store the decorators.
+    * @param {boolean} shouldRemoveDecorators - A flag to indicate whether decorators hould be removed.
+    * @param {string} decl - declaration on which decorator has been applied
+    * @param {string} property - property on which decorator has been applied
+    */
+    static processDecorators(namespace, decorators, decoratorDict, shouldRemoveDecorators,decl,property) {
```

Comment:
Is there another name that's more descriptive for this function? `process` doesn't tell me anything about what this function does unfortunately.

I also think that this should have a `@private` annotation in the JSDoc (this will hide it from the api.txt file and TypeScript definitions file).

Reply:
updated in latest commits